ballerina/schemaClient.bal

```
@@ -0,0 +1,300 @@
+        }
+
+        return routines;
+    }
+
+    # Retrieves information relevant to the provided routine in the database.
+    #
+    # + name - The name of the routine
+    # + return - An 'sql:RoutineDefinition' with the relevant routine information or an `sql:Error`
+    isolated remote function getRoutineInfo(string name) returns sql:RoutineDefinition|sql:Error {
+        record {}|sql:Error routine = self.dbClient->queryRow(
+            `SELECT ROUTINE_TYPE, DATA_TYPE FROM INFORMATION_SCHEMA.ROUTINES 
+             WHERE ROUTINE_NAME = ${name};`
+        );
+
+        if routine is sql:NoRowsError {
+            return error sql:NoRowsError("Selected Routine does not exist or the user does not have privilages of viewing it");
+        } else if routine is sql:Error {
+            return routine;
+        } else {
+            sql:ParameterDefinition[] parameterList = [];
+
+            stream<sql:ParameterDefinition, sql:Error?> paramResults = self.dbClient->query(
+                `SELECT P.PARAMETER_MODE, P.PARAMETER_NAME, P.DATA_TYPE
+                FROM INFORMATION_SCHEMA.PARAMETERS AS P
+                JOIN INFORMATION_SCHEMA.ROUTINES AS R
+                ON P.SPECIFIC_NAME = R.SPECIFIC_NAME
+                WHERE (P.SPECIFIC_SCHEMA = ${self.database} AND R.ROUTINE_NAME = ${name});`
+            );
+            do {
+                check from sql:ParameterDefinition parameters in paramResults
+                    do {
+                        sql:ParameterDefinition 'parameter = {
+                            mode: <sql:ParameterMode>parameters["PARAMETER_MODE"],
+                            name: <string>parameters["PARAMETER_NAME"],
+                            'type: <string>parameters["DATA_TYPE"]
+                        };
+                        parameterList.push('parameter);
+                    };
+            } on fail error e {
+                return error sql:Error("Error - recieved sql data is of type SQL:Error", cause = e);
```

Not a meaningful error message. Better to return the error itself